lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -1454,3 +1455,27 @@ bool CGHeroInstance::hasVisions(const CGObjectInstance * target, const int subty
 
 	return (distance < visionsRange) && (target->pos.z == pos.z);
 }
+
+bool CGHeroInstance::isMissionCritical(const CMap* map) const
```

Comment:
As for triggeredEvents it may be IObjectInterface::cb->getMapHeader()->triggeredEvents